pkgs/development/haskell-modules/non-hackage-packages.nix

```diff
@@ -7,6 +7,10 @@
 # files.
 self: super: {
 
+  # Used by maintainers/scripts/regenerate-hackage-packages.sh, and generated
+  # from the latest master instead of the current version on Hackage.
+  cabal2nix-latest = self.callPackage ./cabal2nix-latest.nix { };
```

Comment:
If this `cabal2nix-latest` will only be used by the `regenerate-hackage-packages.sh`, then we could probably just use IFD with `callCabal2nix` to generate it here based on the current `master` in the cabal2nix repo.

That might be simpler than having to keep around `cabal2nix-latest.nix` and `update-cabal2nix-latest.sh`.

Or maybe you could even just use `callCabal2nix` directly in the `regenerate-hackage-packages.sh` file.

Reply:
I don't think we want to use the current master of the cabal2nix repo because then the version isn't pinned anymore and the current hackage-packages.nix cannot be easily reproduced. We could IFD from a pinned tarball of the cabal2nix repo, but I didn't see a good reason not have it prebuilt by Hydra and instead built by every user of the script.

Reply:
> I don't think we want to use the current master of the cabal2nix repo because then the version isn't pinned anymore and the current hackage-packages.nix cannot be easily reproduced. 

I was under the impression that peti would sometimes update `hackage-packages.nix` with different (in-developement?) versions of `cabal2nix`.

If that is the case, I don't know if we really get anything from keeping this cabal2nix-latest pinned here in nixpkgs.

Unless we have some sort of automatic update, it will be somewhat annoying to keep `cabal2nix-latest` up-to-date manually (or semi-automatically).

Reply:
All statements I found declare that the file is generated from the current master of the nixos/cabal2nix, and this seems to be at least commonly the case.
Merging NixOS/cabal2nix#456 would ensure that this is always the case, as I'd argue it absolutely should be for purposes of reproducibility.
The changes in that PR also ensure that `cabal2nix-latest` is kept in line with the version of cabal2nix used to generate `hackage-packages.nix`.
The other PR, NixOS/cabal2nix#457, also updates the `cabal2nix-latest` to the latest master but does not ensure that the checked out version of cabal2nix used to generate `hackage-packages.nix` actually corresponds to that - it would be on the person making the commit to not forget to push the appropriate version to Github.

Reply:
It looks like NixOS/cabal2nix#457 and NixOS/cabal2nix#456 have been closed.

Does that change the approach you want to take in this PR?

Reply:
I'll leave it up to someone else to fork (or change directly, if they are a maintainer) this PR if they find it useful and implement commit parsing, as I do not find myself willing to put something like that into nixpkgs.